Source/Diagnostics/ReducedDiags/ChargeOnEB.cpp

```diff
@@ -0,0 +1,203 @@
+        const amrex::Array4<const amrex::Real> & dSy_fraction_arr = eb_area_fraction[1]->array(mfi);
+        const amrex::Array4<const amrex::Real> & dSz_fraction_arr = eb_area_fraction[2]->array(mfi);
+
+        amrex::ParallelFor( box,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+
+                // Only cells that are partially covered do contribute to the integral
+                if (eb_flag_arr(i,j,k).isRegular() || eb_flag_arr(i,j,k).isCovered()) return;
+
+                // Find nodal point which is outside of the EB
+                // (eb_normal points towards the *interior* of the EB)
+                int const i_n = (eb_bnd_normal_arr(i,j,k,0) > 0)? i : i+1;
+                int const j_n = (eb_bnd_normal_arr(i,j,k,1) > 0)? j : j+1;
+                int const k_n = (eb_bnd_normal_arr(i,j,k,2) > 0)? k : k+1;
+
+                // Find cell-centered point which is outside of the EB
+                // (eb_normal points towards the *interior* of the EB)
+                int i_c = i;
+                if ((eb_bnd_normal_arr(i,j,k,0)>0) && (eb_bnd_cent_arr(i,j,k,0)<=0)) i_c -= 1;
+                if ((eb_bnd_normal_arr(i,j,k,0)<0) && (eb_bnd_cent_arr(i,j,k,0)>=0)) i_c += 1;
+                int j_c = j;
+                if ((eb_bnd_normal_arr(i,j,k,1)>0) && (eb_bnd_cent_arr(i,j,k,1)<=0)) j_c -= 1;
+                if ((eb_bnd_normal_arr(i,j,k,1)<0) && (eb_bnd_cent_arr(i,j,k,1)>=0)) j_c += 1;
+                int k_c = k;
+                if ((eb_bnd_normal_arr(i,j,k,2)>0) && (eb_bnd_cent_arr(i,j,k,2)<=0)) k_c -= 1;
+                if ((eb_bnd_normal_arr(i,j,k,2)<0) && (eb_bnd_cent_arr(i,j,k,2)>=0)) k_c += 1;
+
+                // Compute contribution to the surface integral $\int dS \cdot E$)
+                amrex::Real local_integral_contribution = 0;
+                local_integral_contribution += Ex_arr(i_c,j_n,k_n)*dSx*(dSx_fraction_arr(i,j,k)-dSx_fraction_arr(i+1,j,k));
+                local_integral_contribution += Ey_arr(i_n,j_c,k_n)*dSy*(dSy_fraction_arr(i,j,k)-dSy_fraction_arr(i,j+1,k));
+                local_integral_contribution += Ez_arr(i_n,j_n,k_c)*dSz*(dSz_fraction_arr(i,j,k)-dSz_fraction_arr(i,j,k+1));
+
+                // Add weighting if requested by user
+                if (do_parser_weighting) {
+                    // Get the 3D position of the centroid of surface element
+                    amrex::Real x = (i + 0.5_rt + eb_bnd_cent_arr(i,j,k,0))*dx[0] + real_box.lo(0);
+                    amrex::Real y = (j + 0.5_rt + eb_bnd_cent_arr(i,j,k,1))*dx[1] + real_box.lo(1);
+                    amrex::Real z = (k + 0.5_rt + eb_bnd_cent_arr(i,j,k,2))*dx[2] + real_box.lo(2);
+                    // Apply weighting
+                    local_integral_contribution *= fun_weightingparser(x, y, z);
+                }
+
+                amrex::Gpu::Atomic::AddNoRet( surface_integral_pointer, local_integral_contribution );
```

Comment:
The PR looks good to me and is ready to go.

I just add an inline comment, as per [Dave's request](https://github.com/ECP-WarpX/WarpX/pull/3648/files#r1113447139):
```suggestion
                // Given that only a tiny fraction of the cells have a non-zero contribution
                // (the ones that intersect with the EB), it is not clear whether ReduceOpSum
                // or AtomicAdd would be faster. However, the implementation with AtomicAdd is easier.
                amrex::Gpu::Atomic::AddNoRet( surface_integral_pointer, local_integral_contribution );
```